core/src/main/java/io/aiven/klaw/helpers/db/rdbms/InsertDataJdbc.java

```diff
@@ -162,30 +162,26 @@ public synchronized Map<String, String> insertIntoRequestConnector(
     return hashMap;
   }
 
-  public synchronized String insertIntoTopicSOT(List<Topic> topics) {
-    Set<Topic> existingTopicIds = new HashSet<>();
+  public synchronized CRUDResponse<Topic> insertIntoTopicSOT(List<Topic> topics) {
+    Set<Integer> existingTopicIds = new HashSet<>();
     topicRepo
         .findAllById(
             topics.stream()
                 .filter(t -> !t.isExistingTopic())
                 .map(t -> new TopicID(t.getTopicid(), t.getTenantId()))
                 .collect(Collectors.toList()))
-        .forEach(existingTopicIds::add);
+        .forEach(entry -> existingTopicIds.add(entry.getTopicid()));
 
-    for (Topic topic : existingTopicIds) {
-      log.debug("insertIntoTopicSOT {}", topic.getTopicname());
-      topic.setTopicid(getNextTopicRequestId("TOPIC_ID", topic.getTenantId()));
-      topicRepo.save(topic);
-    }
     for (Topic topic : topics) {
```

Comment:
I still think the old logic is better than the two new for loops 

        topics.forEach(
        topic -> {
          log.debug("insertIntoTopicSOT {}", topic.getTopicname());
          if (!topic.isExistingTopic()) {
            TopicID topicID = new TopicID();
            topicID.setTenantId(topic.getTenantId());
            topicID.setTopicid(topic.getTopicid());
            if (topicRepo.existsById(topicID)) {
              topic.setTopicid(getNextTopicRequestId("TOPIC_ID", topic.getTenantId()));
            }
          }
          topicRepo.save(topic);
        });

Reply:
So per Sergeys PR https://github.com/Aiven-Open/klaw/pull/1561 this is supposed to be a neater less intensive call now, by using the findByID.

Am happy to go with the consensus but just want to highlight that first.